gax-httpjson/src/main/java/com/google/api/gax/httpjson/InstantiatingHttpJsonChannelProvider.java

```diff
@@ -160,6 +169,20 @@ public TransportChannelProvider withCredentials(Credentials credentials) {
         "InstantiatingHttpJsonChannelProvider doesn't need credentials");
   }
 
+  HttpTransport createHttpTransport() throws IOException {
+    if (mtlsProvider.useMtlsClientCertificate()) {
+      try {
+        KeyStore mtlsKeyStore = mtlsProvider.getKeyStore();
+        if (mtlsKeyStore != null) {
+          return new NetHttpTransport.Builder().trustCertificates(null, mtlsKeyStore, "").build();
+        }
+      } catch (GeneralSecurityException e) {
+        throw new IOException(e.toString());
```

Comment:
same as above. If you must convert a security exception into an IOException do it as late as possible.

Important: throw new IOException(e); instead of throw new IOException(e.toString()); so it doesn't lose the original stack trace.

Reply:
Done. Wrapped the exception in the caller instead of here, and used `new IOException(e)`.